test/modules/fundingManager/bondingCurveFundingManager/RedeemingB_CurveFundingManagerBase.t.sol

```diff
@@ -0,0 +1,461 @@
+            address(bondingCurveFundingManager.formula()),
+            formula,
+            "Formula has not been set correctly"
+        );
+        assertEq(
+            bondingCurveFundingManager.buyFee(),
+            BUY_FEE,
+            "Buy fee has not been set correctly"
+        );
+        assertEq(
+            bondingCurveFundingManager.buyIsOpen(),
+            BUY_IS_OPEN,
+            "Buy-is-open has not been set correctly"
+        );
+        assertEq(
+            bondingCurveFundingManager.buyFee(),
+            SELL_FEE,
+            "Sell fee has not been set correctly"
+        );
+        assertEq(
+            bondingCurveFundingManager.buyIsOpen(),
+            SELL_IS_OPEN,
+            "Sell-is-open has not been set correctly"
+        );
+    }
+
+    function testReinitFails() public override {
+        vm.expectRevert(OZErrors.Initializable__AlreadyInitialized);
+        bondingCurveFundingManager.init(_orchestrator, _METADATA, abi.encode());
+    }
+
+    /* Test sellingIsEnabled modifier
+        ├── when sell is not open
+        │       └── it should revert
+        └── when sell is open
+                └── it should not revert (tested in sellOrder tests)
+
+    */
+    function testSellingIsEnabled_FailsIfSellNotOpen() public {
+        vm.prank(owner_address);
+        bondingCurveFundingManager.closeSell();
+
+        vm.prank(non_owner_address);
+        vm.expectRevert(
+            IRedeemingBondingCurveFundingManagerBase
+                .RedeemingBondingCurveFundingManager__SellingFunctionaltiesClosed
+                .selector
+        );
+        bondingCurveFundingManager.sellOrderFor(non_owner_address, 100);
+    }
+
+    // test modifier on sellOrderFor function
+
+    function testPassingModifiersOnSellOrderFor(uint sellAmount) public {
+        // Setup
+        vm.assume(sellAmount > 0);
+
+        address seller = makeAddr("seller");
+        address receiver = makeAddr("receiver");
+
+        _prepareSellConditions(seller, sellAmount);
+
+        // Pre-checks
+        uint bondingCurveBalanceBefore =
+            _token.balanceOf(address(bondingCurveFundingManager));
+        uint receiverBalanceBefore = _token.balanceOf(receiver);
+        assertEq(_token.balanceOf(seller), 0);
+        assertEq(_token.balanceOf(receiver), 0);
+        assertEq(bondingCurveFundingManager.balanceOf(seller), sellAmount);
+
+        // Execution
+        vm.prank(seller);
+        bondingCurveFundingManager.sellOrderFor(receiver, sellAmount);
+
+        // Post-checks
+        uint redeemAmount = _token.balanceOf(receiver) - receiverBalanceBefore;
+        assertEq(
+            _token.balanceOf(address(bondingCurveFundingManager)),
+            (bondingCurveBalanceBefore - redeemAmount)
+        );
+        assertEq(bondingCurveFundingManager.balanceOf(seller), 0);
+        assertEq(_token.balanceOf(seller), 0);
+    }
+
+    /* Test sellOrder and _sellOrder function
+        ├── when the sell amount is 0
```

Comment:
@0xNuggan did you write this tree yourself?

Reply:
Yes. But apparently missed one statement to be controlled for...